### DISCUSSION TITLE
Event handlers with virtual items in useListNavigation

### DISCUSSION THREAD

**Discussion by what-is-a-crow** [floating-ui/floating-ui#1781] [React] [answered]
I've had good luck adopting the `useListNavigation` hook to build custom selects for an internal design system; however, I'm stuck on the autocomplete / typeahead functionality. I've enabled `virtual` mode, which works but now my items' event handlers (`onKeyDown`, `onClick`, etc) are no longer called. In practice, this means that the text input maintains focus while navigating the menu items but hitting enter does not select the active item.

I played around with the `useCustomLogic` concept with no success but perhaps there is something more I need to do.

Any help is appreciated!

**Comment by atomiks** [comment, marked as answer]
If you check the autocomplete example you can see you need to check `onKeyDown` Enter key on the input/reference element. The `activeIndex` tells you which item has "virtual focus" so you can react similarly

**Comment by what-is-a-crow** [comment]
That makes sense, thank you. I completely missed the set of examples, those are already proving to be very helpful.
